Description: Fix peer certificate verification
 LWP::Protocol::HTTPS disabled peer certificate verification in case
 only hostname verification was disabled.
Origin: vendor
Bug: https://github.com/libwww-perl/lwp-protocol-https/pull/14
Bug-Debian: https://bugs.debian.org/746579
Forwarded: not-needed
Author: Steffen Ullrich <coyote.frank@gmx.net>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2020-07-18

--- a/lib/LWP/Protocol/https.pm
+++ b/lib/LWP/Protocol/https.pm
@@ -20,7 +20,11 @@
 	$ssl_opts{SSL_verifycn_scheme} = 'www';
     }
     else {
-	$ssl_opts{SSL_verify_mode} = 0;
+	if ( $Net::HTTPS::SSL_SOCKET_CLASS eq 'Net::SSL' ) {
+		$ssl_opts{SSL_verifycn_scheme} = '';
+	} else {
+		$ssl_opts{SSL_verifycn_scheme} = 'none';
+	}
     }
     if ($ssl_opts{SSL_verify_mode}) {
 	unless (exists $ssl_opts{SSL_ca_file} || exists $ssl_opts{SSL_ca_path}) {
